### ISSUE TITLE
Mapproxy 1.9, TMS: wrong capabilities and demo not working if combined with WMTS

### ISSUE THREAD

**Issue by anneb** [mapproxy/mapproxy#239]
I believe I found a problem in the Mapproxy 1.9 tms demo for the case where the configuration contains both WMTS and TMS services for MERCATOR layers.

In version 1.8 the TMS demo generates TMS requests like:
http://servername/mapproxy/myconfig/tms/1.0.0/osm/EPSG900913/0/0/1.png

Using the same configuration in version 1.9, the TMS capabilities look like:

```
<?xml version="1.0"?>
    <TileMapService version="1.0.0">
    <Title>MapProxy WMS</Title>
    <Abstract></Abstract>
    <TileMaps>
        <TileMap title="OpenStreetmap"
                 srs="OSGEO:41001"
                 profile="global-mercator"
href="http://localhost/mapproxy/osmtmsproblem/tms/1.0.0/osm/GLOBAL_MERCATOR" />
    </TileMaps>
</TileMapService>

```
and the the TMS demo generates TMS requests like:
http://servername/mapproxy/myconfig/tms/1.0.0/osm/GLOBAL_MERCATOR/0/0/1.png

However, Mapproxy TMS does not support these requests and result in 404 responses with the following error message:
```
<?xml version="1.0"?>
<TileMapServerError>
    <Message>unknown layer: osm</Message>
</TileMapServerError>
```

The TMS requests containing /GLOBAL_MERCATOR/ are generated only if a WMTS service is included in the configuration.

The config that I am using to reproduce the problem in Mapproxy 1.9:

```
services:
  #sets up how to make the source data available
  demo:
  tms:
  wmts:
  wms:

    #srs sets the coordinate reference systems as which you want to make your data available. MapProxy reprojects the source data very well to these projections.
    srs: ['EPSG:900913','EPSG:3857']
    image_formats: ['image/png']

layers:
  #sets up which layers you want to make available using the services above. You can add many, but let's stick to osm data here.
  - name: osm
    title: OpenStreetmap
    sources: [osm_cache]

caches:
  #setup the cache for the open streetmap tiles. This cache is used by the layer above.
  osm_cache:
    disable_storage: true
    sources: [osm_tiles]

sources:
   osm_tiles:
     #the osm_tiles source refers to the openstreetmap.org tiles. These will be downloaded upon request (if not already cached) and served by MapProxy
     type: tile
     url: http://b.tile.openstreetmap.org/%(tms_path)s.%(format)s
     grid: osm_grid #the grid to use for the osm tiles. This is really important. It is specified below.

grids:
  osm_grid:
    #this srs and origin specify a grid that can be used elsewhere in the configuration. In this example it is used for the osm_tiles source. These settings are correct for openstreetmap.org tiles.
    #The google mercator srs is used (also called EPSG:900913), and the origin of the tiles is north-west). If you get this wrong, you might very well get an all-blue world.
    srs: EPSG:900913
    origin: nw

globals:
  #next are some global configuration options for MapProxy. They mostly explain themselves, or can be looked-up in the MapProxy docs.
  cache:
    # where to store the cached images
    base_dir: '/path_to_cache_dir'
    # where to store lockfiles
    lock_dir: '/path_to_locks/'


```

**Comment by olt** [comment]
The metadata of the tile_layer, which contains the URL path was overwritten. This is now fixed.

**Comment by justb4** [comment]
+1 : found same issue in MP 1.9.0, but indeed fixed in TMS/WMTS demo in MapProxy 1.9.1a0-20160920.
